### ISSUE TITLE
R value is not clamped

### ISSUE THREAD

**Issue by Legrandin** [Varbin/python-poly1305#1]
The ``r`` parameter should have 22 of its bits reset to 0 (specifically, when seen as a number, it should be AND-ed with ``0x0ffffffc0ffffffc0ffffffc0fffffff``).

**Comment by Varbin** [comment]
`r &= 0x0ffffffc0ffffffc0ffffffc0fffffff`?

**Comment by Legrandin** [comment]
Yes, if you see `r` as a 128 bit number, that is a mask that clears the appropriate bits as per specification.

**Comment by Varbin** [comment]
Thank you for reporting this (I should have checked the code more carefully!) and sorry for the late (and at first sparse) response. 

The issue should've been resolved with 99ca1baae692f7957c4cac892da0c1e4de717402.
